src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,76 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private boolean isInitialRead = true;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        if (isInitialRead) {
+            String[] dataLinesFromFile = readFromFile(fromFileName);
+            String resultReport = getReport(dataLinesFromFile);
+            writeToFile(resultReport, toFileName);
+        }
+    }
+
+    static String[] readFromFile(String fileName) {
+        File file = new File(fileName);
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String dataLine = reader.readLine();
+            while (dataLine != null) {
+                stringBuilder.append(dataLine).append(System.lineSeparator());
+                dataLine = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from the file " + fileName, e);
+        }
+        return stringBuilder.toString().split(System.lineSeparator());
+    }
+
+    static String getReport(String[] dataLinesFromFile) {
+        final String splitterFields = ",";
+        final int operationTypeIndex = 0;
+        final int amountIndex = 1;
```

Comment:
**Use constants**
If you have strange strings or numbers in the code, it’s better to declare them as constants. The name of the constant should display this object’s purpose.
https://mate-academy.github.io/jv-program-common-mistakes/java-core/builder-file/work-with-file